database/database_test.go

```diff
@@ -234,3 +234,180 @@ func TestStatesWithStatus(t *testing.T) {
 		}
 	}
 }
+
+// setTests is a struct that is used in testing SetIneligibleDomainStates
+var setTests = []struct {
+	description       string
+	domainStates      []IneligibleDomainState
+	wantStatusReports []string
+}{
+	{
+		"two domains",
+		[]IneligibleDomainState{
+			{Name: "youtube.test", Policy: "bulk-1-year"},
+			{Name: "garron.test", Policy: "bulk-1-year"},
+		},
+		[]string{"Updating 2 entries...", " done.\n"},
+	},
+	{
+		"bulk-18-week",
+		[]IneligibleDomainState{{Name: "gmail.test", Policy: "bulk-18-week"}},
+		[]string{"Updating 1 entries...", " done.\n"},
+	},
+	{
+		"bulk-1-year",
+		[]IneligibleDomainState{{Name: "wikipedia.test", Policy: "bulk-1-year"}},
+		[]string{"Updating 1 entries...", " done.\n"},
+	},
+}
+
+// Test SetIneligibleDomainStates
```

Comment:
nit: elaborate in the comment what aspects of SetIneligibleDomainStates this is testing